Zarr Python 2.3 release post

alimanfoo opened the pull request:
This PR adds a blog post about the Zarr Python 2.3 release.

alimanfoo commented:
Hi @zarr-developers/core-devs, this PR brings in draft content for a blog post about the 2.3 Python release, as well as mentioning some future directions including work on new specs. It's WIP but any suggestions or comments very welcome.

jakirkham commented on _posts/2019-05-02-zarr-2.3-release.md:
```diff
@@ -0,0 +1,134 @@
+Python dictionaries (`dict`) implement. The simplicity of this
+interface means it is relatively straightforward to add support for a
+range of different storage systems. The 2.3 release adds support for
+storage using [SQLite](@@TODO), [Redis](@@TODO), [MongoDB](@@TODO) and
```
```suggestion
storage using [SQLite]( https://zarr.readthedocs.io/en/stable/api/storage.html#zarr.storage.SQLiteStore ), [Redis]( https://zarr.readthedocs.io/en/stable/api/storage.html#zarr.storage.RedisStore ), [MongoDB]( https://zarr.readthedocs.io/en/stable/api/storage.html#zarr.storage.MongoDBStore ) and
```

jakirkham commented on _posts/2019-05-02-zarr-2.3-release.md:
```diff
@@ -0,0 +1,134 @@
+interface means it is relatively straightforward to add support for a
+range of different storage systems. The 2.3 release adds support for
+storage using [SQLite](@@TODO), [Redis](@@TODO), [MongoDB](@@TODO) and
+[Azure Blob Storage](@@TODO).
```
```suggestion
[Azure Blob Storage]( https://zarr.readthedocs.io/en/stable/api/storage.html#zarr.storage.ABSStore ).
```

Note: Sorry for the second comment like this. GitHub didn't allow me to consolidate them.

jakirkham commented on _posts/2019-05-02-zarr-2.3-release.md:
```diff
@@ -0,0 +1,134 @@
+
+Support for other cloud object storage storage services was already
+available via other packages, with Amazon S3 supported via the
+[s3fs](@@TODO) package, and Google Cloud Storage supported via the
```
```suggestion
[s3fs]( http://s3fs.readthedocs.io/en/latest/ ) package, and Google Cloud Storage supported via the
```

jakirkham commented on _posts/2019-05-02-zarr-2.3-release.md:
```diff
@@ -0,0 +1,134 @@
+Support for other cloud object storage storage services was already
+available via other packages, with Amazon S3 supported via the
+[s3fs](@@TODO) package, and Google Cloud Storage supported via the
+[gcsfs](@@TODO) package.
```
```suggestion
[gcsfs]( https://gcsfs.readthedocs.io/en/latest/ ) package.
```

jakirkham commented on _posts/2019-05-02-zarr-2.3-release.md:
```diff
@@ -0,0 +1,134 @@
+Around the same time that development on Zarr was getting started, a
+separate team led by Stephan Saafeld at the Janelia research campus
+was experiencing similar frustrations trying to use HDF5 for storing
+brain imaging data, and developed a software library called
```
Electron microscopy data? Also IDK how I feel about "brain" as it feels non-technical, but I don't know how much technical verbage we want to include here. @axtimwalde, thoughts?

alimanfoo replied:
I was hoping you and/or @axtimwalde could improve this :-)

jakirkham commented on _posts/2019-05-02-zarr-2.3-release.md:
```diff
@@ -0,0 +1,134 @@
+## Further reading
+
+* [Zarr Python 2.3 release notes](@@TODO)
+* anything else...
```
Should we engage readers (e.g. "We could really use your help with X?")?

Also do we want to mention that a few of us will be at SciPy?

alimanfoo replied:
> Should we engage readers (e.g. "We could really use your help with X?")?

Good idea, there's lots of things we could use help with but didn't want to overwhelm, not sure what to highlight. I figured engagement in the new spec development is the really big thing, could emphasise that a bit more. Happy to add anything else you think.

> Also do we want to mention that a few of us will be at SciPy?

Good idea!

jakirkham commented:
Looks pretty good. Sorry I started filling in the links, but then figured you probably want high-level feedback to start. Feel free to dismiss those edit comments if you like. Could use a reaching out section at the end. Also do we want to discuss improvements in Numcodecs?

alimanfoo commented:
> Looks pretty good. Sorry I started filling in the links, but then figured you probably want high-level feedback to start. Feel free to dismiss those edit comments if you like. 

Thanks a lot for those edits, I'm really pressed for time at the moment and any little helps :+1: 

> Could use a reaching out section at the end. 

Sounds good, could you suggest some edits there?

> Also do we want to discuss improvements in Numcodecs?

Would be great to surface work on numcodecs, but maybe as a separate post?

jakirkham commented:
I'll try to follow up with more edits on other points then.

The main reason I ask about Numcodecs is it might be valuable to mention improved buffer protocol support. Though this is both in Numcodecs and Zarr. So would make sense to mention in this blogpost.

alimanfoo commented:
> I'll try to follow up with more edits on other points then.

That would be awesome, thanks.

> The main reason I ask about Numcodecs is it might be valuable to mention improved buffer protocol support. Though this is both in Numcodecs and Zarr. So would make sense to mention in this blogpost.

Cool, sounds good, please feel free to add something on that.

alimanfoo commented:
I've done some editing to fill in all the links and examples, and make it (more-or-less) readable. 

@jakirkham would you like to add something in here about numcodecs work? Happy to hold on to this for a little longer if you can't get to it immediately.

alimanfoo commented:
@zarr-developers/core-devs any comments? 

@rabernat I've embedded a slide from your cloud/xarray/zarr talk, hope that's OK.

@axtimwalde / @funkey there's a section here about making the connection between zarr and n5, please feel free to edit or expand.

jakirkham commented:
> @jakirkham would you like to add something in here about numcodecs work? Happy to hold on to this for a little longer if you can't get to it immediately.

Sorry for the delay. Have pushed a short section on buffer protocol support in Zarr and Numcodecs. Please feel free to edit directly as needed.

rabernat commented:
I think this blog post is excellent and interesting! And In the words of Matt Rocklin, even a lousy blog post is better than no blog post.

I suggest we publish this asap.

jakirkham commented:
Happy to publish as well.

alimanfoo commented:
Thanks @jakirkham, awesome. I've added a small bit of context to the section on buffer support, and also added a final section on scipy. Hope that's all OK, going to publish now.

alimanfoo commented:
[Published](https://zarr-developers.github.io/zarr/python/release/2019/05/23/zarr-2.3-release.html) and [tweeted](https://twitter.com/alimanfoo/status/1139096112139292672) :-)

jakirkham commented:
Thanks for the edits and getting this out there. 😄
